lms/static/scripts/frontend_apps/components/test/LMSFilePicker-test.js

```diff
@@ -62,18 +89,103 @@ describe('LMSFilePicker', () => {
 
   it('fetches files when the dialog first appears', async () => {
     const wrapper = renderFilePicker();
+
     assert.calledWith(fakeApiCall, {
       authToken: 'auth-token',
       path: fakeListFilesApi.path,
     });
-    const returnedFiles = await fakeApiCall.returnValues[0];
-    // Component currently filters out Folders
-    const expectedFiles = returnedFiles.filter(file => file.type !== 'Folder');
+
+    const expectedFiles = await fakeApiCall.returnValues[0];
     wrapper.update();
+
     const fileList = wrapper.find('FileList');
     assert.deepEqual(fileList.prop('files'), expectedFiles);
   });
 
+  it('shows breadcrumbs if `withBreadcrumbs` enabled', async () => {
+    const wrapper = renderFilePicker({ withBreadcrumbs: true });
+
+    // Breadcrumbs will only render once the fetch has completed
+    await fakeApiCall;
+    wrapper.update();
```

Comment:
I recommend using `waitFor` here to make the test less brittle. If additional async steps are introduced after the API call in future this could cause the test to fail, even if everything still works from the user's point of view.

Put another way, whenever a test has to wait for some async activity to complete before checking the rendered output, I think it generally makes sense to write it in a way that is flexible with respect to how many async steps happen.